test/generator_test.dart

```diff
@@ -29,4 +78,58 @@ void main() {
     expect(content, contains('this folder: 1'));
     expect(content, contains('sub-folders: 2'));
   });
+  group('build filters', () {
+    test('build all files (without any filters)', () async {
+      final deps = await collectDeps(rootDir: rootDir);
+      final analyzer = Analyzer(deps);
+
+      final generator = MdGenerator(
+        sourceFolder: analyzer.root,
+        rootDir: rootDir,
+        buildFilters: [],
+      );
+
+      final noGeneratedFiles = await generateFiles(generator);
+
+      expect(rootFile.existsSync(), true);
+      expect(noGeneratedFiles, 3);
+      expect(subfolderFile1.existsSync(), true);
+      expect(subfolderFile2.existsSync(), true);
+    });
+    test('build only root', () async {
+      final deps = await collectDeps(rootDir: rootDir);
+      final analyzer = Analyzer(deps);
+
+      final generator = MdGenerator(
+        sourceFolder: analyzer.root,
+        rootDir: rootDir,
+        buildFilters: [Glob('lib')],
+      );
+
+      final noGeneratedFiles = await generateFiles(generator);
+
+      expect(rootFile.existsSync(), true);
+      expect(noGeneratedFiles, 1);
+      expect(subfolderFile1.existsSync(), false);
+      expect(subfolderFile2.existsSync(), false);
+    });
+
+    test('build root and one subfolder', () async {
+      final deps = await collectDeps(rootDir: rootDir);
```

Comment:
Can you add test and example for the case:

lib
lib/subfolder1
lib/subfolder2/**
lib/subfolder2/a

assuming folder structure:

```
lib
  subfolder1
    a
    b
  subfolder2  
    a
    b
```

Check it generate all subtree for subfolder2, and just subfolder1 for subfolder1.

Reply:
I've implemented this test and added a bunch of other tests to really check if it is behaving right and is covered.